pkg/comp-functions/functions/vshnredis/redis_url.go

```diff
@@ -0,0 +1,80 @@
+package vshnredis
+
+import (
+	"context"
+
+	"github.com/crossplane/crossplane/apis/apiextensions/fn/io/v1alpha1"
+	vshnv1 "github.com/vshn/appcat/v4/apis/vshn/v1"
+	"github.com/vshn/appcat/v4/pkg/comp-functions/runtime"
+	v1 "k8s.io/api/core/v1"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+)
+
+var (
+	// RedisHost is env variable in the connection secret
+	RedisHost = "REDIS_HOST"
+	// RedisUser is env variable in the connection secret
+	RedisUser = "REDIS_USERNAME"
+	// RedisPassword is env variable in the connection secret
+	RedisPassword = "REDIS_PASSWORD"
+	// RedisPort is env variable in the connection secret
+	RedisPort = "REDIS_PORT"
+	// RedisUrl is env variable in the connection secret
+	RedisUrl = "REDIS_URL"
+)
+
+// connectionSecretResourceName is the resource name defined in the composition
+// This name is different from metadata.name of the same resource
+// The value is hardcoded in the composition for each resource and due to crossplane limitation
+// it cannot be matched to the metadata.name
+var connectionSecretResourceName = "connection"
+
+// AddUrlToConnectionDetails changes the desired state of a FunctionIO
+func AddUrlToConnectionDetails(ctx context.Context, iof *runtime.Runtime) runtime.Result {
+	log := controllerruntime.LoggerFrom(ctx)
+
+	comp := &vshnv1.VSHNRedis{}
+	err := iof.Desired.GetComposite(ctx, comp)
+	if err != nil {
+		return runtime.NewFatalErr(ctx, "Cannot get composite from function io", err)
+	}
+
+	// Wait for the next reconciliation in case instance namespace is missing
+	if comp.Status.InstanceNamespace == "" {
```

Comment:
Not sure why you're waiting for this status here?

You don't seem to be reading the value for the secret itself?